context/app/static/js/components/entity-search/MetadataMenu/MetadataMenu.jsx

```diff
@@ -38,16 +39,29 @@ function NotebookMenuItem(props) {
   );
 }
 
-function MetadataMenu({ entityType }) {
+function MetadataMenu({ entityType, results }) {
   const lcPluralType = `${entityType.toLowerCase()}s`;
   const { selectedHits, createNotebook, closeMenu } = useMetadataMenu(lcPluralType);
   const menuID = 'metadata-menu';
 
   const { selectedRows } = useStore();
-  const errorMessage =
-    selectedRows.size > 10
-      ? `You have selected ${selectedRows.size} datasets. Workspaces currently only supports up to 10 datasets. Please unselect datasets.`
-      : null;
+  const protectedRows = useDatasetsAccessLevel(selectedRows.size > 0 ? [...selectedRows] : []).datasets;
+
+  const containsProtectedDataset = protectedRows.length > 0;
+
+  const errorMessages = [];
+
+  if (selectedRows.size > 10) {
+    errorMessages.push(
+      `You have selected ${selectedRows.size} datasets. Workspaces currently only supports up to 10 datasets. Please unselect datasets.`,
+    );
+  }
+
+  if (containsProtectedDataset) {
+    errorMessages.push(
+      'You have selected protected datasets. Workspaces currently only supports published public datasets. Selected protected datasets are shown below.',
+    );
+  }
```

Comment:
Converted errorMessage string to errorMessages array and removed "if", "else if" conditions to store more than one errorMessages and map it out so that the errors are stacked.

![image](https://github.com/hubmapconsortium/portal-ui/assets/45410262/24eb29d0-e50e-4f09-bc3d-b56c2fab4a88)